pkg/api/search.go

```diff
@@ -66,8 +67,50 @@ func (hs *HTTPServer) Search(c *models.ReqContext) response.Response {
 		return response.Error(500, "Search failed", err)
 	}
 
-	c.TimeRequest(metrics.MApiDashboardSearch)
-	return response.JSON(http.StatusOK, searchQuery.Result)
+	defer c.TimeRequest(metrics.MApiDashboardSearch)
+
+	if !c.QueryBool("accesscontrol") {
+		return response.JSON(http.StatusOK, searchQuery.Result)
+	}
+
+	return hs.searchHitsWithMetadata(c, searchQuery.Result)
+}
+
+func (hs *HTTPServer) searchHitsWithMetadata(c *models.ReqContext, hits models.HitList) response.Response {
+	folderUIDs := make(map[string]bool, 0)
+	dashboardUIDs := make(map[string]bool, 0)
+	parentUIDs := make(map[string]bool, 0)
```

Comment:
I think we don't need `parentUIDs` and `parentMeta`. Could we use `foldersUIDs` and `folderMeta` instead?

Reply:
Good point we can reuse the folderUIDs and add any dashbord parent uid to it